includes/admin/notes/class-wc-notes-run-db-update.php

```diff
@@ -0,0 +1,312 @@
+<?php
+/**
+ * WooCommerce: Db update note.
+ *
+ * Adds a note to complete the WooCommerce db update after the upgrade in the WC Admin context.
+ *
+ * @package WooCommerce
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\Jetpack\Constants;
+use \Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+/**
+ * WC_Notes_Run_Db_Update.
+ */
+class WC_Notes_Run_Db_Update {
+	const NOTE_NAME = 'wc-update-db-reminder';
+
+	/**
+	 * Attach hooks.
+	 */
+	public function __construct() {
+		if ( defined( 'DOING_AJAX' ) && DOING_AJAX ) {
+			return;
+		}
+
+		add_action( 'admin_init', array( __CLASS__, 'show_reminder' ) );
+		// If the old notice gets dismissed, also hide this new one.
+		add_action( 'woocommerce_hide_update_notice', array( __CLASS__, 'set_notice_actioned' ) );
+	}
+
+	/**
+	 * Get current notice id from the database.
+	 *
+	 * Retrieves the first notice of this type.
+	 *
+	 * @return int|void Note id or null in case no note was found.
+	 */
+	private static function get_current_notice() {
+		try {
+			$data_store = \WC_Data_Store::load( 'admin-note' );
+		} catch ( Exception $e ) {
+			return;
+		}
+		$note_ids   = $data_store->get_notes_with_name( self::NOTE_NAME );
+
+		if ( empty( $note_ids ) ) {
+			return;
+		}
+
+		return current( $note_ids );
```

Comment:
are we returning the last note_id? lets be more explicit here. WDYT?

Reply:
In either case, there should be max 1 of these notes in the db, so I'm returning the 'current' one. Maybe I should add some cleanup to make sure there are no more instances of this note in the db, then it should not matter?

Reply:
If its usually 1 then I guess current implementation is fine. This is just a nitpick but I find `current` hard to read if don't see any iterations on the list nearby. 
Lets keep the current implementation for now.